### ISSUE TITLE
doxygen modules included with tagfiles not visible

### ISSUE THREAD

**Issue by randaz81** [doxygen/doxygen#9424]
I have two projects, A and B. Each of them generates its own documentation and tagfile. 

The two projects are documented using modules, i.e. each of them has a .dox file which contains some \defgroup xxx, \defgroup yyy etc. The my c++ classes are then documented via \ingroup xxx, \ingroup yyy etc. The name of the groups are identical in both projects, i.e. both A and B contains xxx and yyy groups.

Now, I have I want to create a project C, which includes A and B tagfiles. My expectation would be to have the documentation of C showing in group xxx the classes from A and B which belong to xxx, and in group yyy the classes from A and B which belong to yyy.
So, basically, I want to merge the modules documentation of A and B in a single documentation.
However, while other parts of the documentation are ok (e.g. the list of the files shows files from both A and B projects), I was not able to get the modules working. Indeed the modules page for project C is blank. 

Can you please tell me if what I am trying to do is feasible (and I'm thus doing something wrong in my dox configuration files) or if there are issues / not supported?

**Comment by albert-github** [comment]
To be able to tell what might be wrong:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by randaz81** [comment]
I prepared a small zip to show the issue.
[project.zip](https://github.com/doxygen/doxygen/files/9000704/project.zip)

To test it, you just need to run `doxygen Doxyfile` inside `proj_a`, `proj_b`, and `proj_c` folders.
You'll notice that in _proj_c_ the **modules** documentation does not include _class_a_ and _class_b_ which I was expecting to be obtained from the tagfiles.

I tested it on Doxygen 1.8.17 and 1.9.5.
Thak you

**Comment by albert-github** [comment]
You mentioned that you tested with the 1.9.5 version, I tested with the current master  i.e. 1.9.5 (c50f335cc831356fc4046d70a2174c5649171e1e) and in the `modules` / `group` page I see

![image](https://user-images.githubusercontent.com/5223533/176180003-906109de-83b4-490f-86b0-63174662453c.png)

and in here :

![image](https://user-images.githubusercontent.com/5223533/176179953-4d865cfe-bfa4-4037-813b-760e5c2299ca.png)

In the `class list` we see:

![image](https://user-images.githubusercontent.com/5223533/176180144-3c09156e-9971-4d1a-ba2d-5193ab88e1cd.png)


and when I understand it well, you would expect to see something like:

![image](https://user-images.githubusercontent.com/5223533/176180270-634c250d-0a89-4803-acac-8de014c53931.png)

underneath the classes listing

![image](https://user-images.githubusercontent.com/5223533/176180380-9ab0d0a5-f9b4-4d71-94a7-68d9f59db6c9.png)

on the "Devices title" page as well.

In the tag file from `proj_a` we see:
```
  <compound kind="group">
    <name>all_devices</name>
    <title>Devices title</title>
    <filename>group__all__devices.html</filename>
    <class kind="class">class_a</class>
  </compound>
```
and in the tag file from `proj_c` we see:
```
  <compound kind="group">
    <name>all_devices</name>
    <title>Devices title</title>
    <filename>group__all__devices.html</filename>
    <class kind="class">class_c</class>
  </compound>
```
so actually all "identical" just a different class name.

@doxygen 
- Should for a `group`, in case the `name` and `title` are identical, there be a mention of the classes imported through the tag file as well just like on the class page:
   ```
   class_a [external]
   ``` 
- should there be extra restrictions?
- probably this should also be the case for namespaces, functions, ...

**Comment by randaz81** [comment]
> and when I understand it well, you would expect to see something like ... underneath the classes listing ... on the "Devices title" page as well.

Yes, exactly.
Thank you for testing it.
I the meanwhile I also tried to replace [\defgroup](https://www.doxygen.nl/manual/commands.html#cmddefgroup) with [\addtogroup](https://www.doxygen.nl/manual/commands.html#cmdaddtogroup) command but unsuccessfully (same result)
